tests/test_reference_samplers.py

```diff
@@ -17,16 +17,68 @@
 
 import dimod
 from dwave.system.testing import MockDWaveSampler
-
 import hybrid
 from hybrid.samplers import QPUSubproblemAutoEmbeddingSampler
 from hybrid.reference.kerberos import KerberosSampler
+from hybrid.reference.latticeLNLS import LatticeLNLSSampler
+from hybrid.decomposers import make_origin_embeddings
 from hybrid.reference.pa import (
     EnergyWeightedResampler, ProgressBetaAlongSchedule,
     CalculateAnnealingBetaSchedule, PopulationAnnealing
 )
 
 
+class MockDWaveSamplerGeneralization(MockDWaveSampler):
+    """Extend the `dwave.system.testing.MockDWaveSampler` to Pegasus topology.
+    
+    Adding topology and shape keywords to MockDWaveSampler for this purpose.
+    This function is mirrored in test_decomposers.py
+
+    MockDWaveSampler() in the latest version of dwave-system support these 
+    options, this function is included to support backward compatibility of the
+    dwave-system package.
+    """
+    def __init__(self, broken_nodes=None, topology_type=None, qpu_scale=4, **config):
+        import dwave_networkx as dnx
+    
+        super().__init__(broken_nodes, **config)
+        #An Advantage generation processor, only artificially smaller,
+        #replaces C4 in default MockDWaveSampler
+        if topology_type != 'chimera':
+            self.properties['topology'] = {'type': 'pegasus',
+                                           'shape': [qpu_scale]}
+            qpu_graph = dnx.pegasus_graph(qpu_scale,fabric_only=True)
+        else:
+            self.properties['topology'] = {'type': 'chimera',
+                                           'shape': [qpu_scale,qpu_scale,4]}
+            qpu_graph = dnx.chimera_graph(qpu_scale)
+            
+        #Adjust edge_list, 
+        if broken_nodes is None:
+            self.nodelist = sorted(qpu_graph.nodes)
+            self.edgelist = sorted(tuple(sorted(edge))
+                                   for edge in qpu_graph.edges)
+        else:
+            self.nodelist = sorted(v for v in qpu_graph.nodes
+                                   if v not in broken_nodes)
+            self.edgelist = sorted(tuple(sorted((u, v)))
+                                   for u, v in qpu_graph.edges
+                                   if u not in broken_nodes
+                                   and v not in broken_nodes)
+        self.properties['num_qubits'] = len(qpu_graph)
+        self.properties['qubits'] = self.nodelist
+        self.properties['couplers'] = self.edgelist
+        
+
+class TestLatticeLNLS(unittest.TestCase):
+
+    def test_basic_operation(self):
+        bqm = dimod.BinaryQuadraticModel({(i,j,k) : 0 for i in range(2) for j in range(2) for k in range(2)}, {((0,0,0),(0,0,1)): 1, ((1,1,0),(1,1,1)): 1}, 0, dimod.SPIN)
+        sampleset = LatticeLNLSSampler().sample(
+            bqm=bqm, problem_dims=(2,2,2), qpu_sampler=MockDWaveSampler(), topology='cubic',max_iter=1,
+            qpu_params=dict(chain_strength=2), reject_small_problems=False)
```

Comment:
For some reason this still goes to QPU, and [fails tests](https://app.circleci.com/pipelines/github/dwavesystems/dwave-hybrid/524/workflows/ab8bb5ca-3b40-4d40-a097-7fd139c9a13d/jobs/5445).

Reply:
The MockDWaveSampler() lacks topology information that is essential to the test functioning correctly, thats an oversight.

Reply:
In the latest version of dwave-system, the lack of topology information is addressed, but I can use the placeholder mock sampler to allow greater backward compatibility - that is what I thought I had done.